detectors/aws/eks/eks_resource_detector_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package eks
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bmizerany/assert"
+	"github.com/stretchr/testify/mock"
+
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/semconv"
+)
+
+type MockDetectorUtils struct {
+	mock.Mock
+}
+
+// Mock function for fileExists()
+func (detectorUtils *MockDetectorUtils) fileExists(filename string) bool {
+	args := detectorUtils.Called(filename)
+	return args.Bool(0)
+}
+
+// Mock function for fetchString()
+func (detectorUtils *MockDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+	args := detectorUtils.Called(httpMethod, URL)
+	return args.String(0), args.Error(1)
+}
+
+// Mock function for getContainerID()
+func (detectorUtils *MockDetectorUtils) getContainerID() (string, error) {
+	args := detectorUtils.Called()
+	return args.String(0), args.Error(1)
+}
+
+func TestEks(t *testing.T) {
+
+	detectorUtils := new(MockDetectorUtils)
+
+	// Set EKS resource detector variables
+	k8sTokenPath := "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Line 56-60 can be removed, since it is declaring the same variables as defined in eks_resource_detector.go package, those can be called directly in the tests without re-instantiating them.